packages/calling-stateful-client/src/CallSubscriber.ts

```diff
@@ -248,14 +256,26 @@ export class CallSubscriber {
   /* @conditional-compile-remove(close-captions) */
   private initCaptionSubscriber = (): void => {
     // subscribe to captions here so that we don't call captions when call is not initialized
-    if (this._call.state === 'Connected' && !this._captionsSubscriber) {
+    if (
+      this._call.state === 'Connected' &&
+      !this._TeamsCaptionsSubscriber &&
+      /* @conditional-compile-remove(acs-close-captions) */ !this._CaptionsSubscriber
+    ) {
       if (this._call.feature(Features.Captions).captions.kind === 'TeamsCaptions') {
-        this._captionsSubscriber = new CaptionsSubscriber(
+        this._TeamsCaptionsSubscriber = new TeamsCaptionsSubscriber(
           this._callIdRef,
           this._context,
           this._call.feature(Features.Captions).captions as TeamsCaptions
         );
         this._call.off('stateChanged', this.initCaptionSubscriber);
+      } else {
+        /* @conditional-compile-remove(acs-close-captions) */
+        this._CaptionsSubscriber = new CaptionsSubscriber(
+          this._callIdRef,
+          this._context,
+          this._call.feature(Features.Captions).captions as Captions
+        );
+        this._call.off('stateChanged', this.initCaptionSubscriber);
```

Comment:
this happens at the end of the `if` and `else` clause.
Pull out of the if-else and call afterwards

Reply:
Updated